app/src/components/Fields/TimeField.js

```diff
@@ -3,23 +3,24 @@ import PropTypes from 'prop-types';
 import TextFieldMaterial from '@material-ui/core/TextField';
 
 const TimeField = props => {
-  const { id, label, formik, isRequired } = props;
+  const { id, label, formik, isrequired, onChange, picker } = props;
+  const pickerProps = picker ? props : null;
   return (
     <TextFieldMaterial
-      required={isRequired}
+      required={isrequired}
       id={id}
       name={id}
       label={label}
       type="time"
       helperText={formik.touched[id] ? formik.errors[id] : ''}
       error={formik.touched[id] && Boolean(formik.errors[id])}
       value={formik.values[id]}
-      onChange={formik.handleChange}
+      onChange={onChange || formik.handleChange}
       onBlur={formik.handleBlur}
       autoComplete="on"
       fullWidth
       InputLabelProps={{ shrink: true }}
-      {...props}
+      {...pickerProps}
```

Comment:
i want to make sure that we know that we are no longer spreading props in the case that `picker` is passed, so if there are any uses of this component that relied on props that aren't explicitly called above then those could break if `picker` is passed.

what i might suggest, without really knowing what this is doing, if you wanted to keep the props spread functionality, i'd suggest something like this:
```
   const { id, label, formik, isrequired, onChange, pickerProps } = props; 

InputLabelProps={{ shrink: true }}
{...props}
{...pickerProps}
```

which does the same thing -- any props that are in the pickerProps will override any props that aren't explicitly passed.

Reply:
The logic: if picker then spread props. Doing this resolves isRequired in all cases looking back, so switched isrequired back to isRequired. 

Changed is picker to picker="true" on prop pass in, some material-ui components throw errors on a boolean prop.